#VRML_SIM R2022a utf8
WorldInfo {
}
Viewpoint {
  orientation 0.17668464985469706 0.8232042052751113 0.5395529361639092 0.37221414218415233
  position -4.761478954907338 -1.8917480262209492 3.130032354473169
  follow "Crazyflie"
}
TexturedBackground {
}
TexturedBackgroundLight {
}
Floor {
}
Robot {
  translation 0 0 0.015
  children [
    DEF Multiranger Transform {
      translation 0 0 0.02
      children [
        DistanceSensor {
          rotation 0 0 1 -1.57
          name "range_right"
          lookupTable [
            0 0 0
            2 2000 0
          ]
        }
        DistanceSensor {
          rotation 0 0 1 3.14
          name "range_back"
          lookupTable [
            0 0 0
            2 2000 0
          ]
        }
        DistanceSensor {
          rotation 0 0 1 1.57
          name "range_left"
          lookupTable [
            0 0 0
            2 2000 0
          ]
        }
        DistanceSensor {
          name "range_front"
          lookupTable [
            0 0 0
            2 2000 0
          ]
        }
      ]
    }
    Camera {
      translation 0.03 0 0.01
      fieldOfView 0.87
      width 324
      height 324
      exposure 0.2
      motionBlur 0.2
      noise 0.02
      lens Lens {
        radialCoefficients 0.1 0.1
      }
      lensFlare LensFlare {
      }
    }
    Solid {
      translation 0 0 -0.015
      children [
        DEF battery Shape {
          appearance PBRAppearance {
            baseColor 0.5 0.5 0.6
            metalness 0.1
            emissiveIntensity 0
          }
          geometry Mesh {
            url [
              "../../meshes/battery.stl"
            ]
          }
        }
        DEF battery_holder Shape {
          appearance PBRAppearance {
            baseColor 0 0 0
            metalness 0.2
          }
          geometry Mesh {
            url [
              "../../meshes/battery_holder.stl"
            ]
          }
        }
        DEF motors Shape {
          appearance PBRAppearance {
            baseColor 0.5 0.5 0.5
            emissiveColor 0.4 0.4 0.4
          }
          geometry Mesh {
            url [
              "../../meshes/4_motors.stl"
            ]
          }
        }
        DEF motormounts Shape {
          appearance PBRAppearance {
            transparency 0.1
            metalness 0.2
            emissiveColor 0.5 0.5 0.5
          }
          geometry Mesh {
            url [
              "../../meshes/4_motormounts.stl"
            ]
          }
        }
        DEF pinheader Shape {
          appearance DEF metal PBRAppearance {
            baseColor 0.5 0.5 0.5
            metalness 0.8
            emissiveColor 0.4 0.4 0.4
          }
          geometry Mesh {
            url [
              "../../meshes/2_pinheaders.stl"
            ]
          }
        }
        DEF body Shape {
          appearance DEF PCB PBRAppearance {
            baseColor 0 0 0
            roughness 0.3
            metalness 0.5
          }
          geometry Mesh {
            url [
              "../../meshes/cf_body.stl"
            ]
          }
        }
      ]
      name "body"
    }
    GPS {
    }
    Gyro {
    }
    InertialUnit {
    }
    DEF m1 Propeller {
      shaftAxis 0 0 1
      centerOfThrust 0.031 -0.031 0.008
      thrustConstants -4e-05 0
      torqueConstants 2.4e-06 0
      device RotationalMotor {
        name "m1_motor"
        maxVelocity 600
        maxTorque 30
      }
      slowHelix Solid {
        translation 0.031 -0.031 0.008
        children [
          Shape {
            appearance DEF plastic PBRAppearance {
              baseColor 0 0 0
              metalness 0.3
            }
            geometry Mesh {
              url [
                "../../meshes/ccw_prop.stl"
              ]
            }
          }
        ]
      }
    }
    DEF m2 Propeller {
      shaftAxis 0 0 1
      centerOfThrust -0.031 -0.031 0.008
      thrustConstants 4e-05 0
      torqueConstants 2.4e-06 0
      device RotationalMotor {
        name "m2_motor"
        maxVelocity 600
        maxTorque 30
      }
      slowHelix Solid {
        translation -0.031 -0.031 0.008
        children [
          Shape {
            appearance USE plastic
            geometry Mesh {
              url [
                "../../meshes/cw_prop.stl"
              ]
            }
          }
        ]
      }
    }
    DEF m3 Propeller {
      shaftAxis 0 0 1
      centerOfThrust -0.031 0.031 0.008
      thrustConstants -4e-05 0
      torqueConstants 2.4e-06 0
      device RotationalMotor {
        name "m3_motor"
        maxVelocity 600
        maxTorque 30
      }
      slowHelix Solid {
        translation -0.031 0.031 0.008
        children [
          Shape {
            appearance USE plastic
            geometry Mesh {
              url [
                "../../meshes/ccw_prop.stl"
              ]
            }
          }
        ]
      }
    }
    DEF m4 Propeller {
      shaftAxis 0 0 1
      centerOfThrust 0.031 0.031 0.008
      thrustConstants 4e-05 0
      torqueConstants 2.4e-06 0
      device RotationalMotor {
        name "m4_motor"
        maxVelocity 600
        maxTorque 30
      }
      slowHelix DEF prop Solid {
        translation 0.031 0.031 0.008
        children [
          DEF prop Shape {
            appearance USE plastic
            geometry Mesh {
              url [
                "../../meshes/cw_prop.stl"
              ]
            }
          }
        ]
      }
    }
  ]
  name "Crazyflie"
  boundingObject Cylinder {
    height 0.03
    radius 0.05
  }
  physics Physics {
    density -1
    mass 0.05
  }
  controller "crazyflie_controller"
}
SolidBox {
  translation 2.12 0 0.32
  size 0.1 2 0.5
}
Sassafras {
  translation 4.04 0 -1.39
}
Oak {
  translation 1.38 -2.99 0
}
PottedTree {
  translation 2.06 2.05 0
}
PalmTree {
  translation 0.87 2.07 0
}
Cypress {
  translation 3.51866 -2.77548 0
  rotation 0 0 1 -0.523595307179586
}
LincolnMKZSimple {
  translation -2.80526 2.06474 0.4
  rotation 0 0 1 0.785398
}
